micropython/main.py

```diff
@@ -64,7 +66,21 @@
 imsi = ubirch.get_imsi()
 print("IMSI: " + imsi)
 
-if not ubirch.sim_auth(config["sim"]["pin"]):
+# check if PIN is known
+pin_file = imsi + ".bin"
+pin = ""
+if pin_file in os.listdir('.'):
+    print("loading PIN for " + imsi)
+    with open(pin_file, "rb") as f:
+        pin = f.readline().decode()
+else:
+    print("bootstrapping SIM " + imsi)
+    pin = bootstrap(imsi, KEY_SERVER, config["api"]["upp"])
```

Comment:
This uses config->api->upp, I would have expected something like "password".

Reply:
yes. i did not update the upython config :/ it should be done! ill add a todo